nncf/torch/pruning/base_algo.py

```diff
@@ -255,6 +235,6 @@ def get_stats_for_pruned_modules(self):
 
             layer_info["mask_pr"] = self.pruning_rate_for_mask(minfo)
 
-            stats[str(minfo.module_scope)] = layer_info
+            stats[str(minfo.node_name)] = layer_info
```

Comment:
I returned the `module_scope` attribute, but you should think of having only one `PrunedModuleInfo` object per each pruned module, and storing the affected shared nodes inside it, instead of having one `PrunedModuleInfo` object per each node corresponding to the pruned module and then having to group these based on `module_scope` during stat printing.